app/products/calls/actions/calls.ts

```diff
@@ -0,0 +1,261 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import InCallManager from 'react-native-incall-manager';
+
+import {forceLogoutIfNecessary} from '@actions/remote/session';
+import {fetchUsersByIds} from '@actions/remote/user';
+import {hasMicrophonePermission} from '@app/products/calls/actions/permissions';
+import {
+    getCallsConfig,
+    myselfJoinedCall,
+    myselfLeftCall,
+    setCalls,
+    setChannelEnabled,
+    setConfig,
+    setPluginEnabled,
+    setScreenShareURL,
+    setSpeakerPhone,
+} from '@app/products/calls/state';
+import {
+    Call,
+    CallParticipant,
+    CallsConnection, DefaultCallsConfig,
+    ServerChannelState,
+} from '@app/products/calls/types/calls';
+import Calls from '@constants/calls';
+import NetworkManager from '@managers/network_manager';
+
+import {newConnection} from '../connection/connection';
+
+import type {Client} from '@client/rest';
+import type ClientError from '@client/rest/error';
+import type {IntlShape} from 'react-intl';
+
+// Only exported for tests, not exported from the module index.
+export let connection: CallsConnection | null = null;
+
+export const loadConfig = async (serverUrl: string, force = false) => {
+    const now = Date.now();
+    const config = getCallsConfig(serverUrl);
+
+    if (!force) {
+        const lastRetrievedAt = config.last_retrieved_at || 0;
+        if ((now - lastRetrievedAt) < Calls.RefreshConfigMillis) {
+            return {data: config};
+        }
+    }
+
+    let client: Client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch (error) {
+        return {error};
+    }
+
+    let data;
+    try {
+        data = await client.getCallsConfig();
+    } catch (error) {
+        await forceLogoutIfNecessary(serverUrl, error as ClientError);
+
+        // Reset the config to the default (off) since it looks like Calls is not enabled.
```

Comment:
Good question. I'm not totally sure tbh, I was following what was there in v1 and v2 seems to follow this pattern pretty extensively. Open to revisiting this if there's a concern.

Reply:
Thanks, addressed in b2994f4b1776755a21a876525482ea4853f82d07
The config will be re-requested in the channelInfo `useEffect` in case this one failed. (With the 20 min timeout to prevent hammering the server.)